demo/flutter/lib/main.dart

```diff
@@ -0,0 +1,330 @@
+
+  final TranscriptCallback _transcriptCallback;
+  final ProcessErrorCallback _processErrorCallback;
+  RemoveListener? _removeVoiceProcessorListener;
+  RemoveListener? _removeErrorListener;
+
+  static Future<CheetahManager> create(
+      String accessKey, String modelPath, TranscriptCallback transcriptCallback, ProcessErrorCallback processErrorCallback) async {
+    Cheetah cheetah = await Cheetah.create(accessKey, modelPath);
+    return CheetahManager._(cheetah, transcriptCallback, processErrorCallback);
+  }
+
+  CheetahManager._(this._cheetah, this._transcriptCallback, this._processErrorCallback)
+      : _voiceProcessor = VoiceProcessor.getVoiceProcessor(
+            _cheetah!.frameLength, _cheetah.sampleRate) {
+    if (_voiceProcessor == null) {
+      throw CheetahRuntimeException("flutter_voice_processor not available.");
+    }
+    _removeVoiceProcessorListener =
+        _voiceProcessor!.addListener((buffer) async {
+      List<int> cheetahFrame;
+      try {
+        cheetahFrame = (buffer as List<dynamic>).cast<int>();
+      } on Error {
+        CheetahException castError = CheetahException(
+            "flutter_voice_processor sent an unexpected data type.");
+        _processErrorCallback(castError);
+        return;
+      }
+
+      try {
+        CheetahTranscript? cheetahTranscript = await _cheetah?.process(cheetahFrame);
+        _transcriptCallback(cheetahTranscript?.transcript ?? "");
+
+        if (cheetahTranscript?.isEndpoint ?? false) {
+          CheetahTranscript? cheetahTranscript = await _cheetah?.flush();
+          _transcriptCallback(" " + (cheetahTranscript?.transcript ?? "") + " ");
+        }
+      } on CheetahException catch (error) {
+        _processErrorCallback(error);
+      }
+    });
+
+    _removeErrorListener = _voiceProcessor!.addErrorListener((errorMsg) {
+      CheetahException nativeError = CheetahException(errorMsg as String);
+      _processErrorCallback(nativeError);
+    });
+  }
+
+  Future<void> startProcess() async {
+    if (_cheetah == null || _voiceProcessor == null) {
+      throw CheetahInvalidStateException(
+          "Cannot start Cheetah - resources have already been released");
+    }
+
+    if (await _voiceProcessor?.hasRecordAudioPermission() ?? false) {
+      try {
+        await _voiceProcessor!.start();
+      } on PlatformException {
+        throw CheetahRuntimeException(
+            "Audio engine failed to start. Hardware may not be supported.");
+      }
+    } else {
+      throw CheetahRuntimeException(
+          "User did not give permission to record audio.");
+    }
+  }
+
+  Future<void> stopProcess() async {
+    if (_cheetah == null || _voiceProcessor == null) {
+      throw CheetahInvalidStateException(
+          "Cannot start Cheetah - resources have already been released");
+    }
+
+    if (_voiceProcessor?.isRecording ?? false) {
+      await _voiceProcessor!.stop();
+
+      CheetahTranscript? cheetahTranscript = await _cheetah?.flush();
```

Comment:
I believe since you're already checked for null, you can use `_cheetah!.flush()`